identity-actor/src/actor/errors.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use libp2p::request_response::OutboundFailure;
+
+use crate::didcomm::ThreadId;
+
+/// The `Result` type for the actor.
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// Errors that can occur during actor execution.
+#[non_exhaustive]
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+  #[non_exhaustive]
+  #[error("transport error during {0}")]
+  TransportError(&'static str, #[source] libp2p::TransportError<std::io::Error>),
+  #[error("invalid endpoint")]
+  InvalidEndpoint,
+  #[non_exhaustive]
+  #[error("failure during sending an outbound request and receiving the response")]
+  OutboundFailure(#[source] OutboundFailure),
+  #[error("unexpected request `{0}`")]
+  UnexpectedRequest(String),
+  #[error("handler invocation error: {0}")]
+  HandlerInvocationError(String),
+  #[error("hook invocation error: {0}")]
+  HookInvocationError(String),
+  #[non_exhaustive]
+  #[error("{location} serialization failed during {context} due to: {error_message}")]
+  SerializationFailure {
+    location: ErrorLocation,
+    context: String,
+    error_message: String,
+  },
+  #[error("{location} deserialization failed during {context} due to: {error_message}")]
+  DeserializationFailure {
+    location: ErrorLocation,
+    context: String,
+    error_message: String,
+  },
+  #[error("thread with id `{0}` not found")]
+  ThreadNotFound(ThreadId),
+  #[error("awaiting message timed out on thread `{0}`")]
+  AwaitTimeout(ThreadId),
+  #[error("actor was shutdown")]
+  Shutdown,
+  #[error("actor identity missing")]
+  IdentityMissing,
+}
+
+/// Errors that can occur on the remote actor.
+#[derive(Debug, thiserror::Error, serde::Serialize, serde::Deserialize)]
+pub enum RemoteSendError {
```

Comment:
Are we sure we don't want to mark this as `#[non_exhaustive]`? I know everything in this crate will be marked as unstable, but if we already don't know whether more variants might be needed we should prepare for that now.